src/engine/surface_collision.h

```diff
@@ -46,6 +46,12 @@ s32 f32_find_wall_collision(f32 *xPtr, f32 *yPtr, f32 *zPtr, f32 offsetY, f32 ra
 s32 find_wall_collisions(struct WallCollisionData *colData);
 void resolve_and_return_wall_collisions(Vec3f pos, f32 offset, f32 radius, struct WallCollisionData *collisionData);
 f32 find_ceil(f32 posX, f32 posY, f32 posZ, struct Surface **pceil);
+
+// Finds the ceiling from a vec3f and a minimum height (with 3 unit vertical buffer).
+ALWAYS_INLINE f32 find_mario_ceil(Vec3f pos, f32 height, struct Surface **ceil) {
+    return find_ceil(pos[0], (MAX(height, pos[1]) + 3.0f), pos[2], ceil);
```

Comment:
```suggestion
    return find_ceil(pos[0], MAX(height, pos[1]) + 3.0f, pos[2], ceil);
```
easier to tell whats going on